mobile/lib/features/text_correction/data/data_sources/mistake_local_data_source.dart

```diff
@@ -0,0 +1,75 @@
+import 'package:flutter/cupertino.dart';
+import 'package:mobile/core/data/database.dart';
+import 'package:mobile/core/error/exceptions.dart';
+import 'package:moor/ffi.dart';
+
+abstract class MistakeLocalDataSource {
+  /// Gets the cached list of [Mistake] of a specific correctiion (student and text).
+  ///
+  /// Returns an empty list if no [Mistake] is cache.
+  ///
+  /// Throws [CacheException] if something wrong happens.
+  Future<List<MistakeModel>> getMistakesFromCacheOfCorrection({
```

Comment:
I got a little confused by this name

Did you mean `getCorrectionMistakesFromCache`?

Reply:
yes